app/src/views/features/rules/RuleEditor/RuleEditor.js

```diff
@@ -76,6 +77,12 @@ const RuleEditor = (props) => {
     }
   }, [state?.source, MODE]);
 
+  useEffect(() => {
+    if (state?.source) {
+      trackRuleEditorViewed(state?.source, currentlySelectedRuleData?.ruleType);
+    }
+  }, [state?.source, currentlySelectedRuleData?.ruleType]);
+
```

Comment:
There is a `RuleBuilder`component in this file which also fires `trackRuleEditorViewed`. Please take a look at that as well , and lets fire the event only from one instance.